OFEP-flag-metadata-to-applications.md

```diff
@@ -0,0 +1,80 @@
+## OFEP - Expose flag metadata to applications
+
+## State: DRAFTING
+
+OpenFeature specification focuses on the evaluation of feature flags. To further enhance the application author
+experience, this proposal focuses on exposing flag metadata through dedicated client methods.
+
+
+## Background
+
+At the time of writing this document, OpenFeature API supports metadata for providers and client. With a similar 
+approach, it is desirable to expose flag metadata to the application layer.
+
+This metadata then can be used by application authors for various purposes such as validations, tooling and testing.
+For example, cross-validations of flags used by an application using unit tests without the need for evaluation API 
+or a connection to the flag management system.
+
+## Proposal
+
+The content of metadata payload can vary from provider to provider. However, from OpenFeature specification perspective,
+it should be possible to enforce the following, 
+
+- flag key 
+- flag type 
```

Comment:
Implying that flag value should not be returned within the metadata? It may be prudent to explicitly state this